scripts/dev/simple_init.py

```diff
@@ -36,19 +36,22 @@
 
 
 def wait_for_all_processes(ref_pid):
-    child_errors = False
+    ref_status = None
+    first_non_zero_child_status = 0
     while True:
         try:
             # Wait for any child
             child_pid, child_status = os.waitpid(-1, 0)
-            #print("ripped a child", child_pid, child_status)
-            # Check its status (the reference child is not considered here since it will be checked later on)
-            if child_pid != ref_pid and child_status != 0:
-                child_errors = True
+            print("ripped a child", child_pid, child_status)
```

Comment:
:man_facepalming: I shouldn't have uncommented this line. I guess it's not critical, but it would be good to comment it back at the next PR on version/2.5